fairlearn/postprocessing/_thresholder.py

```diff
@@ -0,0 +1,105 @@
+
+# try to get it to work now
+from utils._common import _get_soft_predictions
+from utils._input_validation import _validate_and_reformat_input
+from postprocessing._constants import (
+    BASE_ESTIMATOR_NONE_ERROR_MESSAGE,
+    BASE_ESTIMATOR_NOT_FITTED_WARNING)
+
+
+# how it should be
+# from ..utils._common import _get_soft_predictions
+# from ..utils._input_validation import _validate_and_reformat_input
+# from ._constants import (
+#     BASE_ESTIMATOR_NONE_ERROR_MESSAGE,
+#     BASE_ESTIMATOR_NOT_FITTED_WARNING)
+
+
+class Thresholder(BaseEstimator, MetaEstimatorMixin):
+    r"""Create my own description here
+
+    Parameters
+
+    References
+    ----------
+
+    """
+
+    def __init__(self, estimator, threshold_dict, prefit=False,
+                 predict_method='deprecated'):
+        self.estimator = estimator
+        self.threshold_dict = threshold_dict
+        self.prefit = prefit
+        self.predict_method = predict_method
+
+    def fit(self, X, y, **kwargs):
+        """Fit the estimator.
+
+        If `prefit` is set to `True` then the base estimator is kept as is.
+        Otherwise it is fitted from the provided arguments.
+        """
+        if self.estimator is None:
+            raise ValueError(BASE_ESTIMATOR_NONE_ERROR_MESSAGE)
+
+        if self.predict_method == "deprecated":
+            warn(
+                "'predict_method' default value is changed from 'predict' to "
+                "'auto'. Explicitly pass `predict_method='predict' to "
+                "replicate the old behavior, or pass `predict_method='auto' "
+                "or other valid values to silence this warning.",
+                FutureWarning,
+            )
+            self._predict_method = "predict"
+        else:
+            self._predict_method = self.predict_method
+
+        if not self.prefit:
+            self.estimator_ = clone(self.estimator).fit(X, y, **kwargs)
+        else:
+            try:
+                check_is_fitted(self.estimator)
+            except NotFittedError:
+                warn(BASE_ESTIMATOR_NOT_FITTED_WARNING.format(type(self).__name__))
+            self.estimator_ = self.estimator
+        return self
+
+    def predict(self, X, *, sensitive_features):
+        """
+        Predict stuff, write better explanation later"""
+
+        check_is_fitted(self)
+
+        # get soft predictions
+        base_predictions = np.array(
+            _get_soft_predictions(self.estimator_, X, self._predict_method)
+        )
+
+        # validate and reformat input
+        _, base_predictions_vector, sensitive_feature_vector, _ = _validate_and_reformat_input(
+            X, y=base_predictions, sensitive_features=sensitive_features, expect_y=True,
+            enforce_binary_labels=False)
+
+        final_predictions = 0.0*base_predictions_vector
+
+        for a, threshold in self.threshold_dict.items():
+
+            operation = ThresholdOperation('>', threshold)

```

Comment:
So we don't allow < thresholds? Wouldn't the threshold dict have the full specification of the threshold operation? Maybe I'm misunderstanding the input type of `threshold_dict`, though, which is entirely possible given how long it's been since I worked on it. If you're not sure yourself I'll be happy to dig in a bit more, just lmk!

Reply:
Consider the following example `threshold_dict`: {'A' : .6}

Now, for group 'A' we predict 1 if the output > .6, and 0 otherwise. It seems a bit redundant (and prone to confusion) to also allow threshold operations going the other way to be included in the dict.

Reply:
I see. I think what confused me is that I used the term `threshold_dict` elsewhere in the past as a dict mapping the sensitive feature group names to `ThresholdOperation` objects, not just to floating point numbers. In my case, I can make this richer in the sense that I get to specify not just the float, but also the operator, so that's where my confusion stems from.

Taking a step back, I'm still not convinced we should give up on such a capability (thresholds in both directions). What do others think @fairlearn/fairlearn-maintainers ?

Reply:
I don't have a very strong opinion on this. I think defaulting`'>'` suffices for most use cases and simplifies the API. I'm not opposed to having the capability, provided it does not add too much complexity for how the user needs to supply the thresholds.

What would you suggest the API to look like if we go for both directions, @romanlutz? E.g., should the user supply a dict of `ThresholdOperation` objects, a dict of tuples `(float, operator)`, a dict of floats and a separate argument for `operator`, or perhaps something else entirely?